### ISSUE TITLE
CERES issues

### ISSUE THREAD

**Issue by cdcseacave** [facebookresearch/vggsfm#93]
I followed the install readme (`source install.sh`) but any reconstruction fails with:
```
(vggsfm_tmp) ubuntu@150-136-33-157:~/vggsfm$ python demo.py SCENE_DIR=examples/kitchen/
Poselib is available
Poselib is available
/home/ubuntu/vggsfm/demo.py:17: UserWarning:
The version_base parameter is not specified.
Please specify a compatability version level, or None.
Will assume defaults for version 1.1
  @hydra.main(config_path="cfgs/", config_name="demo")
/home/ubuntu/miniconda3/envs/vggsfm_tmp/lib/python3.10/site-packages/hydra/_internal/hydra.py:119: UserWarning: Future Hydra versions will no longer change working directory at job runtime by default.
See https://hydra.cc/docs/1.2/upgrades/1.1_to_1.2/changes_to_job_working_dir/ for more information.
  ret = run_job(
Model Config: model_name: vggsfm_v2_0_0
seed: 0
img_size: 1024
debug: false
center_order: false
mixed_precision: fp16
extract_color: true
filter_invalid_frame: true
comple_nonvis: true
query_frame_num: 3
robust_refine: 2
BA_iters: 1
fine_tracking: true
camera_type: SIMPLE_PINHOLE
load_gt: false
viz_visualize: false
gr_visualize: false
fmat_thres: 4.0
max_reproj_error: 4.0
init_max_reproj_error: 4.0
max_query_pts: 2048
dense_depth: false
avg_pose: true
shared_camera: false
save_to_disk: true
SCENE_DIR: examples/kitchen/
resume_ckpt: ckpt/vggsfm_v2_0_0.bin
auto_download_ckpt: true
query_method: aliked
use_poselib: true
shift_point2d_to_original_res: false
make_reproj_video: false
visual_tracks: false
visual_query_points: false
visual_dense_point_cloud: false
query_by_midpoint: false
query_by_interval: false
concat_extra_points: false
extra_pt_pixel_interval: -1
extra_by_neighbor: -1
MODEL:
  _target_: vggsfm.models.VGGSfM
  TRACK:
    _target_: vggsfm.models.TrackerPredictor
    efficient_corr: false
    COARSE:
      stride: 4
      down_ratio: 2
      FEATURENET:
        _target_: vggsfm.models.BasicEncoder
      PREDICTOR:
        _target_: vggsfm.models.BaseTrackerPredictor
    FINE:
      FEATURENET:
        _target_: vggsfm.models.ShallowEncoder
      PREDICTOR:
        _target_: vggsfm.models.BaseTrackerPredictor
        depth: 4
        corr_levels: 3
        corr_radius: 3
        latent_dim: 32
        hidden_size: 256
        fine: true
        use_spaceatt: false
  CAMERA:
    _target_: vggsfm.models.CameraPredictor
  TRIANGULAE:
    _target_: vggsfm.models.Triangulator

Building VGGSfM
Using cache found in /home/ubuntu/.cache/torch/hub/facebookresearch_dinov2_main
/home/ubuntu/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/swiglu_ffn.py:51: UserWarning: xFormers is not available (SwiGLU)
  warnings.warn("xFormers is not available (SwiGLU)")
/home/ubuntu/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/attention.py:33: UserWarning: xFormers is not available (Attention)
  warnings.warn("xFormers is not available (Attention)")
/home/ubuntu/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/block.py:40: UserWarning: xFormers is not available (Block)
  warnings.warn("xFormers is not available (Block)")
[2024-12-31 16:16:24,445][dinov2][INFO] - using MLP layer as FFN
VGGSfM built successfully
Data size of Sequence: 1
Run Sparse Reconstruction for Scene kitchen
[W NNPACK.cpp:64] Could not initialize NNPACK! Reason: Unsupported hardware.
Predicting tracks with query_index = 16
Predicting tracks with query_index = 1
Predicting tracks with query_index = 7
Error executing job with overrides: ['SCENE_DIR=examples/kitchen/']
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ubuntu/vggsfm/demo.py", line 71, in demo_fn
    predictions = vggsfm_runner.run(
  File "/home/ubuntu/vggsfm/vggsfm/runners/runner.py", line 228, in run
    predictions = self.sparse_reconstruct(
  File "/home/ubuntu/vggsfm/vggsfm/runners/runner.py", line 500, in sparse_reconstruct
    ) = self.triangulator(
  File "/home/ubuntu/miniconda3/envs/vggsfm_tmp/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/vggsfm_tmp/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/vggsfm/vggsfm/models/triangulator.py", line 149, in forward
    ) = init_BA(
  File "/home/ubuntu/vggsfm/vggsfm/utils/triangulation.py", line 210, in init_BA
    ba_options = prepare_ba_options()
  File "/home/ubuntu/vggsfm/vggsfm/utils/triangulation_helpers.py", line 628, in prepare_ba_options
    ba_options_tmp.solver_options.function_tolerance *= 10
TypeError: Unable to convert function return value to a Python type! The signature was
	(self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```
pls help

**Comment by jackdaus** [comment]
# tldr 
Ensure that `pyceres` version `2.3` is installed in your environment (not the newer `2.4` version). To do that, run this command,
```bash
pip install pyceres==2.3
```

# Explanation
I believe this is related to issue https://github.com/facebookresearch/vggsfm/issues/85

It looks like this issue is caused by the version of `pyceres` being used. A new version (`2.4`) was recently published on November 28, 2024 (https://pypi.org/project/pyceres/#history). But the `install.sh` script in this vggsfm repo does not specify a version, so it seems like they might be "out of sync" or something.

Based on [this comment from COLMAP authors](https://github.com/colmap/colmap/issues/2937#issuecomment-2497890815), this error might also be fixed by upgrading to `pycolmap` version `3.11.0`? Not sure about that. But downgrading `pyceres` fixed the error message to me.

**Comment by cdcseacave** [comment]
this solved the error, thank you
